### PR TITLE
History of viewed cars connected back with front

### PR DESCRIPTION
**Pull request by Aleksandr-Shulha-dev**
General view of the component on the page: 
image
https://gyazo.com/47b5aa304f162ec3927819c5eaa7cef6

### CODE REVIEW

**Comment by Aleksandr-Shulha-dev** [comment]
Pagination data: https://gyazo.com/47b5aa304f162ec3927819c5eaa7cef6

**Comment by Aleksandr-Shulha-dev** [comment]
I'm sorry about a large number of modified files. When i was creating the client`s side and the server side, I did not take into account many details (the dark side filled my mind 👾). And when I understood it, it was too late( It was the last time, I promise.

**Comment by oleksandr-danylchenko** [comment]
@Aleksandr-Shulha-dev 
Why did you use the centering of the card labels? In design, we use left aligning
image

**Comment by oleksandr-danylchenko** [comment]
> 

I don't understand what happens in the video :eye
Do you load the new cars on "View more"?

**Comment by oleksandr-danylchenko** [comment]
Why there are 2 cars with the same model+complectation combination on the page?
image

**Comment by oleksandr-danylchenko** [comment]
> Pagination data: https://gyazo.com/47b5aa304f162ec3927819c5eaa7cef6

I don't understand what happens in the video :eye
Do you load the new cars on "View more"?

**Comment by oleksandr-danylchenko** [comment]
> I'm sorry about a large number of modified files. When i was creating the client`s side and the server side, I did not take into account many details (the dark side filled my mind space_invader). And when I understood it, it was too late( It was the last time, I promise.

It's never late to create 2 new branches and cherry pick changes to them separately

File: backend/src/helpers/viewed-cars/viewed-cars.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const formatDtoForResponse = (
+  data: ViewedCarPrismaDto,
+): ViewedCarResponseDto => ({
+  brand: data.brand.name,
+  model: data.name,
+  complectation: data.complectations[0] ? data.complectations[0].name : '',
+  year: data.year_end
+    ? `${data.year_start} - ${data.year_end}`
+    : String(data.year_start),
```
Comment (oleksandr-danylchenko) [inline backend/src/helpers/viewed-cars/viewed-cars.ts]: The `${data.year_start}` would be preferred more than the `String` constuctor

File: frontend/src/components/viewed-cars/viewed-cars.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    userId: '0cdfe5ca-256f-49e4-855f-f438a4fac3c9',
+    skip: '0',
+    take: String(takeCars),
+  };
+
+  const { data, isLoading } = useGetHistoryOfViwedCarsQuery(params);
+
+  const getMoreCars = async (): Promise<void> => {
+    setTakeCars(takeCars + 30);
+  };
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/viewed-cars/viewed-cars.tsx]: `useGetHistoryOfViwedCarsQuery` would perform the requests which will contain the already downloaded cars. So the pagination concept isn't really working here

File: frontend/src/components/viewed-cars/viewed-cars.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import type { GetViewedCarsResponse } from '@autoline/shared';
+
+const ViewedCars: FC = () => {
+  const [takeCars, setTakeCars] = useState(6);
+
+  const params = {
+    userId: '0cdfe5ca-256f-49e4-855f-f438a4fac3c9',
+    skip: '0',
+    take: String(takeCars),
+  };
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/viewed-cars/viewed-cars.tsx]: This object will be recreated on each rerender and the `useGetHistoryOfViwedCarsQuery` will be restarted on each rerender as well

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]
The infinity scroll concept or the pagination won't work here bc on each request we'll get the already fetched data in the response

**Comment by Aleksandr-Shulha-dev** [comment]
> @Aleksandr-Shulha-dev Why did you use the centering of the card labels? In design, we use left aligning image

In order for the button "load more" was in the center
image

**Comment by oleksandr-danylchenko** [comment]
> In order for the button "load more" was in the center image

Not sure how the centering of the container element should impact the cards. This should be fixed and the centering should be applied only the `Load more` button

**Comment by Aleksandr-Shulha-dev** [comment]
> Why there are 2 cars with the same model+complectation combination on the page? image

In the database we have a lot of identical cars with different id. And I used old data. Now it's looks like this. 
image

**Comment by Aleksandr-Shulha-dev** [comment]
> @Aleksandr-Shulha-dev Why did you use the centering of the card labels? In design, we use left aligning image



> Why there are 2 cars with the same model+complectation combination on the page? image



> Why there are 2 cars with the same model+complectation combination on the page? image



> > Pagination data: https://gyazo.com/47b5aa304f162ec3927819c5eaa7cef6
> 
> I don't understand what happens in the video :eye Do you load the new cars on "View more"?


I mistakenly added another video.

https://gyazo.com/a7528e611f1cb3a09c1da5b679d7eff4

**Comment by Aleksandr-Shulha-dev** [comment]
image

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**